Ifpack2: error when both FastILU and Sacado enabled

If Ifpack2, Sacado and ShyLU_NodeFastILU are enabled together, there are a couple of build errors:

```
In file included from ompBuild/packages/stokhos/src/Ifpack2_Details_FastILU_Base_MP_Vector_16_OpenMP.cpp:51:
Trilinos/packages/ifpack2/src/Ifpack2_Details_FastILU_Base_def.hpp: In instantiation of ‘Ifpack2::Details::FastILU_Base<Scalar, LocalOrdinal, GlobalOrdinal, Node>::Params::Params(const Teuchos::ParameterList&, std::string) [with Scalar = Sacado::MP::Vector<Stokhos::StaticFixedStorage<int, double, 16, Kokkos::OpenMP> >; LocalOrdinal = int; GlobalOrdinal = int; Node = Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::OpenMP>; std::string = std::__cxx11::basic_string<char>]’:
ompBuild/packages/stokhos/src/Ifpack2_Details_FastILU_Base_MP_Vector_16_OpenMP.cpp:55:1:   required from here
Trilinos/packages/ifpack2/src/Ifpack2_Details_FastILU_Base_def.hpp:508:13: error: cannot convert ‘const Sacado::MP::Vector<Stokhos::StaticFixedStorage<int, double, 16, Kokkos::OpenMP> >’ to ‘double’ in assignment
  508 |       omega = pL.get<magnitude>("damping factor");
Trilinos/packages/ifpack2/src/Ifpack2_Details_FastILU_Base_def.hpp:519:13: error: cannot convert ‘const Sacado::MP::Vector<Stokhos::StaticFixedStorage<int, double, 16, Kokkos::OpenMP> >’ to ‘double’ in assignment
  519 |       shift = pL.get<magnitude>("shift");
```

At first glance, it looks like the Ifpack2 wrapper for FastILU is assuming that Scalar can be implicitly converted to double.

--- a/packages/ifpack2/src/Ifpack2_Details_FastILU_Base_def.hpp
+++ b/packages/ifpack2/src/Ifpack2_Details_FastILU_Base_def.hpp
@@ -1,4 +1,4 @@
- /*@HEADER
+/*@HEADER
 // ***********************************************************************
 //
 //       Ifpack2: Templated Object-Oriented Algebraic Preconditioner Package
@@ -428,7 +428,6 @@ FastILU_Base<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
 Params::Params(const Teuchos::ParameterList& pL, std::string precType)
 {
   *this = getDefaults();
-  typedef typename Teuchos::ScalarTraits<Scalar>::magnitudeType magnitude;
   //For each parameter, check that if the parameter exists, it has the right type
   //Then get the value and sanity check it 
   //If the parameter doesn't exist, leave it as default (from Params::getDefaults())
@@ -499,27 +498,19 @@ Params::Params(const Teuchos::ParameterList& pL, std::string precType)
     }
     CHECK_VALUE("level", level, level < 0, "must be nonnegative");
   }
-  //"damping factor" aka omega -- try both double and magnitude as type
   if(pL.isParameter("damping factor"))
   {
     if(pL.isType<double>("damping factor"))
       omega = pL.get<double>("damping factor");
-    else if(pL.isType<magnitude>("damping factor"))
-      omega = pL.get<magnitude>("damping factor");
     else
-      TYPE_ERROR("damping factor", "double or magnitude_type");
-    CHECK_VALUE("damping factor", omega, omega <= 0 || omega > 1, "must be in the range (0, 1]");
+      TYPE_ERROR("damping factor", "double");
   }
-  //"shift" -- also try both double and magnitude
   if(pL.isParameter("shift"))
   {
     if(pL.isType<double>("shift"))
       shift = pL.get<double>("shift");
-    else if(pL.isType<magnitude>("shift"))
-      shift = pL.get<magnitude>("shift");
     else
-      TYPE_ERROR("shift", "double or magnitude_type");
-    //no hard requirements for shift value so don't
+      TYPE_ERROR("shift", "double");
   }
   //"guess" aka guessFlag
   if(pL.isParameter("guess"))

--- a/packages/shylu/shylu_node/fastilu/src/shylu_fastilu.hpp
+++ b/packages/shylu/shylu_node/fastilu/src/shylu_fastilu.hpp
@@ -1750,7 +1750,8 @@ class FastILUPrec
         void apply(ScalarArray &x, ScalarArray &y)
         {
             Kokkos::Timer timer;
-            const Scalar one  = STS::one();
+            const Scalar one(1.0);
+            const Scalar minus_one(-1.0);
 
             //required to prevent contamination of the input.
             ParCopyFunctor<Ordinal, Scalar, ExecSpace> parCopyFunctor(xTemp, x);
@@ -1853,13 +1854,13 @@ class FastILUPrec
                                 // > y = x
                                 Kokkos::parallel_for(Kokkos::RangePolicy<ExecSpace>(0, nRows), copy_x2y);
                                 // > y = y - L*x_old
-                                KokkosSparse::spmv("N", -one, crsmatL, x2d_old, one, y2d);
+                                KokkosSparse::spmv("N", minus_one, crsmatL, x2d_old, one, y2d);
                             } else {
                                 // x_old = x_old - L*y
                                 // > x_old = x
                                 Kokkos::parallel_for(Kokkos::RangePolicy<ExecSpace>(0, nRows), copy_x2xold);
                                 // > x_old = x_old - L*y
-                                KokkosSparse::spmv("N", -one, crsmatL, y2d, one, x2d_old);
+                                KokkosSparse::spmv("N", minus_one, crsmatL, y2d, one, x2d_old);
 
                                 if (i == nTrisol-1) {
                                     // y = x_old
@@ -1891,7 +1892,7 @@ class FastILUPrec
                                 // > x = y
                                 Kokkos::parallel_for(Kokkos::RangePolicy<ExecSpace>(0, nRows), copy_y2x);
                                 // > x = x - U*x_old
-                                KokkosSparse::spmv("N", -one, crsmatU, x2d_old, one, x2d);
+                                KokkosSparse::spmv("N", minus_one, crsmatU, x2d_old, one, x2d);
                                 // > scale x = inv(diag(U))*x
                                 Kokkos::parallel_for(Kokkos::RangePolicy<ExecSpace>(0, nRows), scal_x);
                             } else {
@@ -1899,7 +1900,7 @@ class FastILUPrec
                                 // > xold = y
                                 Kokkos::parallel_for(Kokkos::RangePolicy<ExecSpace>(0, nRows), copy_y2xold);
                                 // > x = x - U*x_old
-                                KokkosSparse::spmv("N", -one, crsmatU, x2d, one, x2d_old);
+                                KokkosSparse::spmv("N", minus_one, crsmatU, x2d, one, x2d_old);
                                 // > scale x = inv(diag(U))*x
                                 Kokkos::parallel_for(Kokkos::RangePolicy<ExecSpace>(0, nRows), scal_xold);
 

